src/MyPlot/EventListener.php

```diff
@@ -234,56 +234,57 @@ public function onEntityMotion(EntityMotionEvent $event) {
 	 * @param PlayerMoveEvent $event
 	 */
 	public function onPlayerMove(PlayerMoveEvent $event) {
-		if($event->isCancelled()) {
+		if($event->isCancelled() or $event->getFrom()->distance($event->getTo()) < 0.01) {
 			return;
 		}
-		if($event->getFrom()->floor()->equals($event->getTo()->floor())) return; // Save hell-a-lot of performance by ignoring movements on the same block, for example rotating the head
 
 		$levelName = $event->getPlayer()->getLevel()->getName();
 		if (!$this->plugin->isLevelLoaded($levelName))
 			return;
 
 		$plot = $this->plugin->getPlotByPosition($event->getTo());
 		$plotFrom = $this->plugin->getPlotByPosition($event->getFrom());
-		if ($plot !== null and $plot !== $plotFrom) {
+		if($plot instanceof Plot and $plotFrom === null) {
 			$this->plugin->getServer()->getPluginManager()->callEvent($ev = new MyPlotPlayerEnterPlotEvent($this->plugin, $event->getPlayer()->getName(), $plot, $event->getPlayer()));
 			if($ev->isCancelled()) {
 				$event->setCancelled();
 				return;
 			}
+		} elseif($plotFrom instanceof Plot and $plot === null) {
+			$this->plugin->getServer()->getPluginManager()->callEvent($ev = new MyPlotPlayerLeavePlotEvent($this->plugin, $event->getPlayer()->getName(), $plot, $event->getPlayer()));
+			if($ev->isCancelled()) {
+				$event->setCancelled();
+				return;
+			}
+		}
 
+		if($plot !== null) {
 			if($plot->isDenied($event->getPlayer()->getName())) {
 				$event->setCancelled();
 				return;
 			}
 
 			$plotName = TextFormat::GREEN . $plot;
-			$popup = $this->plugin->getLanguage()->translateString("popup", [$plotName]);
-			if(strpos($plot,"-0")) { // Jason, tell me WTF this is - strpos needs a string. $plot is not a string.
+			$popup    = $this->plugin->getLanguage()->translateString("popup", [$plotName]);
+			if(strpos($plot, "-0")) {
```

Comment:
So, you are still strpos'ing an object. great work

Reply:
Look into the object’s functions before criticizing. Php allows this under conditions.

Reply:
Only if an object allows casting into string.